src/crush/CrushCompiler.cc

```diff
@@ -88,7 +88,7 @@ static void print_rule_name(ostream& out, int t, CrushWrapper &crush)
 static void print_fixedpoint(ostream& out, int i)
 {
   char s[20];
-  snprintf(s, sizeof(s), "%.3f", (float)i / (float)0x10000);
+  snprintf(s, sizeof(s), "%.5f", (float)i / (float)0x10000);
```

Comment:
```
2020-12-11T07:37:59.623 INFO:tasks.workunit.client.0.smithi037.stdout:--- td/crush-choose-args/map-one-more.txt 2020-12-11 07:37:59.603708149 +0000
2020-12-11T07:37:59.623 INFO:tasks.workunit.client.0.smithi037.stdout:+++ /home/ubuntu/cephtest/clone.client.0/src/test/crush/crush-choose-args-expected-one-more-3.txt 2020-12-11 07:37:38.129035943 +0000
2020-12-11T07:37:59.623 INFO:tasks.workunit.client.0.smithi037.stdout:@@ -29,18 +29,18 @@
2020-12-11T07:37:59.623 INFO:tasks.workunit.client.0.smithi037.stdout: # buckets
2020-12-11T07:37:59.624 INFO:tasks.workunit.client.0.smithi037.stdout: host HOST {
2020-12-11T07:37:59.624 INFO:tasks.workunit.client.0.smithi037.stdout:  id -2           # do not change unnecessarily
2020-12-11T07:37:59.624 INFO:tasks.workunit.client.0.smithi037.stdout:- # weight 6.00000
2020-12-11T07:37:59.624 INFO:tasks.workunit.client.0.smithi037.stdout:+ # weight 6.000
2020-12-11T07:37:59.624 INFO:tasks.workunit.client.0.smithi037.stdout:  alg straw2
2020-12-11T07:37:59.625 INFO:tasks.workunit.client.0.smithi037.stdout:  hash 0  # rjenkins1
2020-12-11T07:37:59.625 INFO:tasks.workunit.client.0.smithi037.stdout:- item osd.0 weight 3.00000
2020-12-11T07:37:59.625 INFO:tasks.workunit.client.0.smithi037.stdout:- item osd.1 weight 3.00000
2020-12-11T07:37:59.625 INFO:tasks.workunit.client.0.smithi037.stdout:+ item osd.0 weight 3.000
2020-12-11T07:37:59.625 INFO:tasks.workunit.client.0.smithi037.stdout:+ item osd.1 weight 3.000
2020-12-11T07:37:59.626 INFO:tasks.workunit.client.0.smithi037.stdout: }
2020-12-11T07:37:59.626 INFO:tasks.workunit.client.0.smithi037.stdout: root default {
2020-12-11T07:37:59.626 INFO:tasks.workunit.client.0.smithi037.stdout:  id -1           # do not change unnecessarily
2020-12-11T07:37:59.626 INFO:tasks.workunit.client.0.smithi037.stdout:- # weight 6.00000
2020-12-11T07:37:59.627 INFO:tasks.workunit.client.0.smithi037.stdout:+ # weight 6.000
2020-12-11T07:37:59.627 INFO:tasks.workunit.client.0.smithi037.stdout:  alg straw2
2020-12-11T07:37:59.627 INFO:tasks.workunit.client.0.smithi037.stdout:  hash 0  # rjenkins1
2020-12-11T07:37:59.627 INFO:tasks.workunit.client.0.smithi037.stdout:- item HOST weight 6.00000
2020-12-11T07:37:59.627 INFO:tasks.workunit.client.0.smithi037.stdout:+ item HOST weight 6.000
2020-12-11T07:37:59.628 INFO:tasks.workunit.client.0.smithi037.stdout: }
2020-12-11T07:37:59.628 INFO:tasks.workunit.client.0.smithi037.stdout:
2020-12-11T07:37:59.628 INFO:tasks.workunit.client.0.smithi037.stdout: # rules
2020-12-11T07:37:59.628 INFO:tasks.workunit.client.0.smithi037.stdout:@@ -59,16 +59,16 @@
2020-12-11T07:37:59.628 INFO:tasks.workunit.client.0.smithi037.stdout:   {
2020-12-11T07:37:59.629 INFO:tasks.workunit.client.0.smithi037.stdout:     bucket_id -1
2020-12-11T07:37:59.629 INFO:tasks.workunit.client.0.smithi037.stdout:     weight_set [
2020-12-11T07:37:59.629 INFO:tasks.workunit.client.0.smithi037.stdout:-      [ 5.00000 ]
2020-12-11T07:37:59.629 INFO:tasks.workunit.client.0.smithi037.stdout:-      [ 5.00000 ]
2020-12-11T07:37:59.629 INFO:tasks.workunit.client.0.smithi037.stdout:+      [ 5.000 ]
2020-12-11T07:37:59.630 INFO:tasks.workunit.client.0.smithi037.stdout:+      [ 5.000 ]
2020-12-11T07:37:59.630 INFO:tasks.workunit.client.0.smithi037.stdout:     ]
2020-12-11T07:37:59.630 INFO:tasks.workunit.client.0.smithi037.stdout:     ids [ -10 ]
2020-12-11T07:37:59.630 INFO:tasks.workunit.client.0.smithi037.stdout:   }
2020-12-11T07:37:59.631 INFO:tasks.workunit.client.0.smithi037.stdout:   {
2020-12-11T07:37:59.631 INFO:tasks.workunit.client.0.smithi037.stdout:     bucket_id -2
2020-12-11T07:37:59.631 INFO:tasks.workunit.client.0.smithi037.stdout:     weight_set [
2020-12-11T07:37:59.631 INFO:tasks.workunit.client.0.smithi037.stdout:-      [ 2.00000 3.00000 ]
2020-12-11T07:37:59.631 INFO:tasks.workunit.client.0.smithi037.stdout:-      [ 2.00000 3.00000 ]
2020-12-11T07:37:59.632 INFO:tasks.workunit.client.0.smithi037.stdout:+      [ 2.000 3.000 ]
2020-12-11T07:37:59.632 INFO:tasks.workunit.client.0.smithi037.stdout:+      [ 2.000 3.000 ]
2020-12-11T07:37:59.632 INFO:tasks.workunit.client.0.smithi037.stdout:     ]
2020-12-11T07:37:59.632 INFO:tasks.workunit.client.0.smithi037.stdout:     ids [ -20 1 ]
2020-12-11T07:37:59.632 INFO:tasks.workunit.client.0.smithi037.stdout:   }
2020-12-11T07:37:59.633 INFO:tasks.workunit.client.0.smithi037.stderr:/home/ubuntu/cephtest/clone.client.0/qa/standalone/crush/crush-choose-args.sh:91: TEST_choose_args_update:  return 1
```

see https://pulpito.ceph.com/kchai-2020-12-11_07:09:12-rados-wip-kefu-testing-2020-12-10-1827-distro-basic-smithi/5700652/

@pdvian could you update `qa/standalone/crush/crush-choose-args.sh` accordingly ?

Reply:
Sure Kefu.